### PR TITLE
Show duration

### PR DESCRIPTION
**Pull request by LennDG**
PR for [the show duration issue](https://github.com/wswebcreation/multiple-cucumber-html-reporter/issues/11).

Adds support for duration display in a readable format.

### CODE REVIEW

**Comment by wswebcreation** [comment]
Hi @LennDG 

Tnx for the PR. This looks already nice. I think we also need to update the other templates (custom-meta templates). I'll also think about a the time in the overview page. Total time can't be calculated because we don't know if the tests have been executed in parallel. 

I only need to check when I have time to dive into this because I have a busy weekend and week, but I'll come back on it. Again tnx!

**Comment by LennDG** [comment]
The custom templates are only for the features overview page (that one might want duration so you can see the duration on the overview page) and for the metadata overview on a feature page, but on that page the duration is already displayed below the chart. 
So unless the features overview page should have the duration, the custom templates don't need to be adjusted.
I indeed do not display total time (although the functionality was in the code) since it is not very useful in many cases.

File: lib/generate-report.js [hunk 1]
```diff
@@ -444,6 +456,33 @@ function generateReport(options) {
...
+
+    function formatDuration(ms) {
+        const timeData = convertMS(ms);
+        var d, h, m, s, ms;
+        ms = timeData.ms ? timeData.ms + "ms" : "";
+        s = timeData.s ? timeData.s + "s" : "";
+        m = timeData.m ? timeData.m + "m" : "";
+        h = timeData.h ? timeData.h + "h" : "";
+        d = timeData.d ? timeData.d + "d" : "";
+        return d + h + m + s + ms;
```
Comment (wswebcreation) [inline lib/generate-report.js]: I've been thinking about this and I wonder what the _best_ layout would be, or yours

image

or this

image

Personally I prefer the last, even though there is no d/h/m/s/ms notation in it.

File: lib/generate-report.js [hunk 2]
```diff
@@ -444,6 +456,33 @@ function generateReport(options) {
             );
         });
     }
+    
+    // From https://gist.github.com/remino/1563878
```
Comment (wswebcreation) [inline lib/generate-report.js]: For calculating times I'd prefer using a library to keep this code as clean as possible, I don't want to make unit test for time calculations, they always go wrong. We can use `momentjs` for it and solve the time calculation with 3 lines of code and a require at the top of the file.

```js
const moment = require('moment');


function formatDuration(ms) {
    return moment.utc(ms).format("HH:mm:ss.SSS")
}
```

File: README.MD [hunk 3]
```diff
@@ -143,6 +143,13 @@ This will disable the log so will **NOT** see this.
 
 You can change the report name to a name you want
 
+### `displayDuration`
```
Comment (wswebcreation) [inline README.MD]: 👍 for adding this as an option!!

File: README.MD [hunk 4]
```diff
@@ -143,6 +143,13 @@ This will disable the log so will **NOT** see this.
...
+- **Type:** `boolean`
+- **Mandatory:** No
+
+If set to true the duration of steps, scenarios and features is displayed on the Feature page in an easily readable format.
+This expects the durations in the report to be in milliseconds, which might result in incorrect durations when using a version of Cucumber that does not report in milliseconds (like v.1).
```
Comment (wswebcreation) [inline README.MD]: 👍

File: lib/generate-report.js (hunk 2 above)
Comment (LennDG) [reply lib/generate-report.js]: Sure, I didn't want to add extra dependencies with my PR. Using a library will save some headaches.

File: lib/generate-report.js (hunk 1 above)
Comment (LennDG) [reply lib/generate-report.js]: This is fine by me, I didn't think of different ways to display and just used the format that the Cucumber Jenkins Plugin uses.

**Comment by LennDG** [comment]
Overall you can just push these changes since I have nothing more to add. Looking forward to using the official package again!

**Pull request merged**